realm/realm-library/src/main/java/io/realm/coroutines/FlowFactory.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2020 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.coroutines;
+
+import javax.annotation.Nonnull;
+
+import io.realm.DynamicRealm;
+import io.realm.DynamicRealmObject;
+import io.realm.Realm;
+import io.realm.RealmList;
+import io.realm.RealmModel;
+import io.realm.RealmObject;
+import io.realm.RealmResults;
+import io.realm.annotations.Beta;
+import kotlinx.coroutines.flow.Flow;
+
+/**
+ * Factory interface for creating Kotlin {@link Flow}s for Realm classes.
+ */
+@Beta
+public interface FlowFactory {
```

Comment:
I think we might want to add the `changeSetsFrom` methods already now to this interface. Even if we don't expose them in this PR, it will prevent a later breaking change?

But then again, the interface is marked Beta.

Thoughts?

Reply:
I will have a look at the current implementation for Rx. I will get back to you once I have a better idea about whether it's doable right away or not.

Reply:
I've added the implementation for all changeset methods and tests for them. I have left all the factory methods markes as "beta".